### ISSUE TITLE
Implement Docker provider

### ISSUE THREAD

**Issue by larsbutler** [checkmate/chessboard#10]
Here are the basic things the docker provider needs to do:

- [ ] Define and provide a catalog
  - TODO: What's in the catalog by default?
- [ ] Convert [custom components](https://github.com/checkmate/chessboard/issues/9) to the appropriate docker artifacts (`Dockerfiles`).
- [ ] Convert `Checkmatefile` blueprint and generate a `docker-compose.yml` to model the application topology.
- [ ] Code to call `docker-compose`.

**Comment by ryandub** [comment]
WRT "Code to call `docker-compose`", you may be able to integrate it directly instead of having to use  a `shell_out` since it is in Python. Not sure if you were already thinking that way, but thought I would throw it in here.

**Comment by larsbutler** [comment]
@ryandub Good point; `docker-compose` is written in Python. I'll look into that.
